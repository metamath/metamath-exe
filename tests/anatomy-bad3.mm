$( anatomy.mm $)

$(
                      PUBLIC DOMAIN DEDICATION

This file is placed in the public domain per the Creative Commons Public
Domain Dedication. http://creativecommons.org/licenses/publicdomain/

Norman Megill - email: nm at alum.mit.edu

$)


$( This file is the introductory formal system example described
   in Chapter 4 of the Metamath book, "Anatomy of a proof". $)

$( Declare the constant symbols we will use $)
$c ( ) -> wff $.
$( Declare the metavariables we will use $)
$v p q r s $.
$( Declare wffs $)
wp $f wff p $.
wq $f wff q $.
wr $f wff r $.
ws $f wff s $.
w2 $a wff ( p -> q ) $.

$( Prove a simple theorem. $)
wnew $p wff ( s -> ( r -> p ) ) $= wr wp w2 w2 $.

