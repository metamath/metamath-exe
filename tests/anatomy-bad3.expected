MM> READ "anatomy-bad3.mm"
Reading source file "anatomy-bad3.mm"... 696 bytes
696 bytes were read into the source buffer.
The source has 8 statements; 1 are $a and 1 are $p.
No errors were found.  However, proofs were not checked.  Type VERIFY PROOF *
if you want to check them.
MM> Continuous scrolling is now in effect.
MM> 0 10%  20%  30%  40%  50%  60%  70%  80%  90% 100%
..................................................
?Error on line 29 of file "anatomy-bad3.mm" at statement 8, label "wnew", type
"$p":
wnew $p wff ( s -> ( r -> p ) ) $= wr wp w2 w2 $.
                                            ^^
At proof step 4, statement "w2" requires 2 hypotheses but the RPN stack
contains only one entry, "w2" (step 3).

